src/NServiceBus.Testing.Fakes/TestablePipelineContext.cs

```diff
@@ -21,6 +21,11 @@ public TestablePipelineContext(IMessageCreator messageCreator = null)
         }
 
         /// <summary>
+        /// A list of all messages sent with a timeout header.
+        /// </summary>
+        public TimeoutMessage<object>[] TimeoutMessages => timeoutMessages.ToArray();
```

Comment:
I like the explicitness. The approach was introduced by a TF around the testing lib. The timeout type is already public there. I think it is simpler for the end users to be able to rely on that type. I'll apply though the other suggestions